### PR TITLE
Get max ammo(pickup) 2

### PR DESCRIPTION
**Pull request by louis1706**

### CODE REVIEW

**Comment by louis1706** [comment]
remake of #1999

File: Exiled.API/Extensions/ItemExtensions.cs [hunk 1]
```diff
@@ -308,5 +319,47 @@ public static uint GetBaseCode(this FirearmType type)
...
+        public static byte GetMaxAmmo(this Pickup pickup)
+        {
+            if (pickup is not FirearmPickup firearm)
+                return 0;
+            byte ammo = GetMaxAmmo(firearm.Info.ItemId.GetFirearmType());
+
+            if (firearm.Status.Flags.HasFlag(FirearmStatusFlags.Chambered))
+                ammo++;
+
+            return ammo += (byte)UnityEngine.Mathf.Clamp(GetAttachmentsValue(firearm, AttachmentParam.MagazineCapacityModifier), byte.MinValue, byte.MaxValue);
```
Comment (iopietro) [inline Exiled.API/Extensions/ItemExtensions.cs:337]:
```suggestion
            return ammo += (byte)UnityEngine.Mathf.Clamp(firearm.GetAttachmentsValue(AttachmentParam.MagazineCapacityModifier), byte.MinValue, byte.MaxValue);
```

File: Exiled.API/Extensions/ItemExtensions.cs [hunk 2]
```diff
@@ -308,5 +319,47 @@ public static uint GetBaseCode(this FirearmType type)
...
+        /// <summary>
+        /// Gets the default Maxammo of a weapon.
+        /// </summary>
+        /// <param name="pickup">The <see cref="Pickup">item</see> that you want to get durability of.</param>
+        /// <returns>Returns the item durability.</returns>
+        public static byte GetMaxAmmo(this Pickup pickup)
+        {
+            if (pickup is not FirearmPickup firearm)
+                return 0;
+            byte ammo = GetMaxAmmo(firearm.Info.ItemId.GetFirearmType());
```
Comment (iopietro) [inline Exiled.API/Extensions/ItemExtensions.cs:332]:
```suggestion

            byte ammo = GetMaxAmmo(firearm.Info.ItemId.GetFirearmType());
```

File: Exiled.API/Extensions/ItemExtensions.cs [hunk 3]
```diff
@@ -308,5 +319,47 @@ public static uint GetBaseCode(this FirearmType type)
...
+            float num = definitionOfParam.DefaultValue;
+
+            foreach (AttachmentIdentifier attachement in attachements)
+            {
+                Attachment attachment = AttachmentsList.FirstOrDefault(x => x.Name == attachement.Name);
+                if (attachment is null || !attachment.TryGetValue((int)attachmentParam, out float paraValue))
+                    continue;
+
+                num = AttachmentsUtils.MixValue(num, paraValue, definitionOfParam.MixingMode);
+            }
```
Comment (iopietro) [inline Exiled.API/Extensions/ItemExtensions.cs:360]:
```suggestion
            foreach (AttachmentIdentifier attachmentIdentifier in attachments)
            {
                Attachment attachment = AttachmentsList.FirstOrDefault(x => x.Name == attachmentIdentifier.Name);

                if (attachment is null || !attachment.TryGetValue((int)attachmentParam, out float paraValue))
                    continue;

                num = AttachmentsUtils.MixValue(num, paraValue, definitionOfParam.MixingMode);
            }
```

File: Exiled.API/Extensions/ItemExtensions.cs [hunk 4]
```diff
@@ -308,5 +319,47 @@ public static uint GetBaseCode(this FirearmType type)
...
+
+        /// <summary>
+        /// Gets the value of an AttachmentParam on a FirearmPickup.
+        /// </summary>
+        /// <param name="firearmPickup">The <see cref="FirearmPickup">pickup</see> that you want to get the value of.</param>
+        /// <param name="attachmentParam">The <see cref="AttachmentParam">AttachmentParam</see> for get the Parameter change you need.</param>
+        /// <returns>Returns the float value.</returns>
+        public static float GetAttachmentsValue(this FirearmPickup firearmPickup, AttachmentParam attachmentParam)
+        {
+            IEnumerable<AttachmentIdentifier> attachements = GetAttachmentIdentifiers(firearmPickup.Info.ItemId.GetFirearmType(), firearmPickup.Status.Attachments);
```
Comment (iopietro) [inline Exiled.API/Extensions/ItemExtensions.cs:348]:
```suggestion
            IEnumerable<AttachmentIdentifier> attachments = GetAttachmentIdentifiers(firearmPickup.Info.ItemId.GetFirearmType(), firearmPickup.Status.Attachments);
```

**Review by iopietro — CHANGES_REQUESTED** [review]
I haven't looked at your algorithm yet.

**Comment by Thundermaker300** [comment]
Fix build errors

**Pull request closed without merging**